src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -264,6 +264,11 @@ export class ExtHostTerminal extends Disposable {
 		return false;
 	}
 
+	public setShellType(shellType: TerminalShellType | number | undefined): void {
+		this._state = { shellType }; // Why is the Type complaining, its the exact same as in proposed.api
```

Comment:
`_state` includes `isInteractedWith`, we must retain that state. Similarly, we must not clobber `shellType` when `isInteractedWith` is set.